llvm/cmake/modules/LLVMExternalProjectUtils.cmake

```diff
@@ -34,6 +34,8 @@ endfunction()
 
 
 # llvm_ExternalProject_Add(name source_dir ...
+#   ENABLE_FORTRAN
+#     External project requires the Flang compiler
```

Comment:
I don't fully understand why you need that, but the fact that we don't need `ENABLE_CPLUSPLUS` makes me think this is smelly. Can you explain?

Reply:
Flang is required to build FlangRuntime. Currently, there is just one file `iso_fortran_env_impl.f90` it is needed for, but there should be no reason why not implement parts of the language's runtime in the language itself as all out C/C++ runtimes do. Additionally, in the future I hope that we can use it to compile the `.mod` files as well.

`ENABLE_FORTRAN` would build Flang before the runtimes and error-out Flang is not part of a bootstrap build. If there no runtime dependent on Fortran then we also don't need to depend on Flang. Adding `ENABLE_CPLUSPLUS` would be kind of pointless because even single runtime we have depends on Clang (unless I am mistaken?).

Reply:
Ah, I see now, `clang` is being added implicitly if you don't specify any required toolchain tools. I don't love adding a new parameter, but after consideration I can't think of any other simple solution.